wp_api_integration_tests/tests/test_plugin_directory.rs

```diff
@@ -1,46 +1,95 @@
-    reqwest::get(url)
-        .await?
-        .json::<QueryPluginResponse>()
-        .await?
-        .plugins
-        .into_iter()
-        .map(|p| Ok(p.slug))
-        .collect()
+use reqwest::Client as ReqwestClient;
+use wp_api::{
+    request::{endpoint::media_endpoint::MediaUploadRequest, RequestMethod, WpNetworkHeaderMap},
+    MediaUploadRequestExecutionError,
+};
+
+#[derive(Debug)]
+pub struct ReqwestExecutor {
+    client: ReqwestClient,
 }
 
-async fn plugin_information(slug: &str) -> Result<PluginInformation, reqwest::Error> {
-    let url = format!(
-        "https://api.wordpress.org/plugins/info/1.2/?action=plugin_information&request[slug]={}&fields=icons",
-        slug
-    );
-    reqwest::get(&url)
-        .await?
-        .json::<PluginInformation>()
-        .await
-        .map_err(Into::into)
+impl ReqwestExecutor {
+    fn request_method(method: RequestMethod) -> http::Method {
+        match method {
+            RequestMethod::GET => reqwest::Method::GET,
+            RequestMethod::POST => reqwest::Method::POST,
+            RequestMethod::PUT => reqwest::Method::PUT,
+            RequestMethod::DELETE => reqwest::Method::DELETE,
+            RequestMethod::HEAD => reqwest::Method::HEAD,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl RequestExecutor for ReqwestExecutor {
+    async fn execute(
+        &self,
+        wp_request: Arc<WpNetworkRequest>,
+    ) -> Result<WpNetworkResponse, RequestExecutionError> {
+        let mut request = self
+            .client
+            .request(
+                Self::request_method(wp_request.method()),
+                wp_request.url().0.as_str(),
+            )
+            .headers(wp_request.header_map().as_header_map());
+        if let Some(body) = wp_request.body() {
+            request = request.body(body.contents());
+        }
+        let mut response =
+            request
+                .send()
+                .await
+                .map_err(|err| RequestExecutionError::RequestExecutionFailed {
+                    status_code: err.status().map(|s| s.as_u16()),
+                    reason: err.to_string(),
+                })?;
+
+        let header_map = std::mem::take(response.headers_mut());
+        Ok(WpNetworkResponse {
+            status_code: response.status().as_u16(),
+            body: response.bytes().await.unwrap().to_vec(),
+            header_map: Arc::new(WpNetworkHeaderMap::new(header_map)),
+        })
+    }
+
+    async fn upload_media(
+        &self,
+        _: Arc<MediaUploadRequest>,
+    ) -> Result<WpNetworkResponse, MediaUploadRequestExecutionError> {
+        unimplemented!("upload_media is not implemented for sending requests to api.wordpress.org")
+    }
+}
+
+fn wordpress_org_api_client() -> WordPressOrgApiClient {
+    let client = ReqwestClient::new();
+    let executor = ReqwestExecutor { client };
+    WordPressOrgApiClient::new(Arc::new(executor))
 }
```

Comment:
```suggestion
fn wordpress_org_api_client() -> WordPressOrgApiClient {
    WordPressOrgApiClient::new(Arc::new(
        wp_api_integration_tests::AsyncWpNetworking::default(),
    ))
}
```

I don't think we need a new executor. You can remove everything  related to `ReqwestExecutor` and replace the `wordpress_org_api_client` function as above and it'll work as expected.